src/Model/User.php

```diff
@@ -143,6 +143,26 @@ public function tear_down() {
 	 */
 	protected function is_private() {
 
+		/**
+		 * Filter whether the user is private.
+		 *
+		 * If true, the user will be considered private without further checks.
+		 *
+		 * @param mixed|null|boolean $is_private If set, return without executing the manual query
+		 *                                       to see if the user has published posts.
+		 * @param User               $this       Instance of the User Model
+		 */
+		$is_private = apply_filters( 'graphql_user_model_is_private', null, $this );
```

Comment:
I think we should probably handle this in the `Model` abstract so all models get this. That's how I handled it here: https://github.com/wp-graphql/wp-graphql/pull/972.

Reply:
I've made updates to @CodeProKid original #972 PR to get it to pass tests, and this now allows the filter to run on all Models. 

If 3rd party code uses that filter, it will be used. If not, it will fall back to the individual Model's `is_private()` method. 

I'll get that merged and released. 

Closing this PR in favor of that one.